agent.py

```diff
@@ -136,8 +137,9 @@ def reload_configs(self, checks_to_reload=set()):
                 jmx_sd_configs = generate_jmx_configs(self._agentConfig, hostname)
         else:
             new_checksd = copy(self._checksd)
-
-            jmx_checks = [check for check in checks_to_reload if check in JMX_CHECKS]
+            auto_conf_is_jmx_checks = get_jmx_checks()
+            all_jmx_checks = JMX_CHECKS + auto_conf_is_jmx_checks
```

Comment:
Had the same thought! Adding to `get_jmx_checks()` and changing to a set, we don't want duplicates.